File: src/main/java/by/kohanova/model/Details.java
package by.kohanova.model;

import javax.persistence.*;
import java.io.Serializable;

@Entity
@Table(name = "details")
public class Details implements Serializable {
    private static final long serialVersionUID = 1L;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Integer id;

    @OneToOne(optional = false, cascade = CascadeType.ALL)
    @JoinColumn(name = "user_id", unique = true, nullable = false)
    public User user;

    @Column
    public String firstname;

    @Column
    public String surname;

    @Column
    public String photo;
}

File: client/src/app/model/details.ts
import {User} from "./user";
export class Details {
    firstname: string;
    surname: string;
    photo: string;
    user: User;

    constructor(name: string, surname: string, photo: string, user: User) {
        this.firstname = name;
        this.surname = surname;
        this.photo = photo;
        this.user = user;
    }
}